satpy/dataset.py

```diff
@@ -46,20 +48,54 @@ def id(self):
         return DatasetID.from_dict(self.attrs)
 
 
-def combine_metadata(*metadata_objects):
+def average_datetimes(dt_list):
+    """Average a series of datetime objects.
+
+    .. note::
+
+        This function assumes all datetime objects are naive and in the same
+        time zone (UTC).
+
+    Args:
+        dt_list (iterable): Datetime objects to average
+
+    Returns: Average datetime as a datetime object
+
+    """
+    if sys.version_info < (3, 3):
+        # timestamp added in python 3.3
+        import time
+
+        def timestamp_func(dt):
+            return time.mktime(dt.timetuple())
+    else:
+        timestamp_func = datetime.timestamp
+
+    total = [timestamp_func(dt) for dt in dt_list]
+    return datetime.fromtimestamp(sum(total) / len(total))
+
+
+def combine_metadata(*metadata_objects, **kwargs):
```

Comment:
why not `def combine_metadata(*metadata_objects, average_times=True):`?

Reply:
Read the comment below. That syntax isn't valid in python 2. I had it that way first and the tests failed. Unless you're saying I should drop python 2 support now... 😜

Reply:
I'm happy with that :)